Gems/ScriptCanvas/Code/Editor/Components/EditorScriptCanvasComponent.cpp

```diff
@@ -30,481 +31,102 @@
-                ->Field("runtimeDataOverrides", &EditorScriptCanvasComponent::m_variableOverrides)
-                ->Field("sourceHandle", &EditorScriptCanvasComponent::m_sourceHandle)
-                ;
-
-            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
-            {
-                editContext->Class<EditorScriptCanvasComponent>("Script Canvas", "The Script Canvas component allows you to add a Script Canvas asset to a component, and have it execute on the specified entity.")
-                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
-                    ->Attribute(AZ::Edit::Attributes::Category, "Scripting")
-                    ->Attribute(AZ::Edit::Attributes::Icon, "Icons/ScriptCanvas/ScriptCanvas.svg")
-                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, "Icons/ScriptCanvas/Viewport/ScriptCanvas.svg")
-                    ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game", 0x232b318c))
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("UI", 0x27ff46b0))
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Level", 0x9aeacc13))
-                    ->Attribute(AZ::Edit::Attributes::HelpPageURL, "https://o3de.org/docs/user-guide/components/reference/scripting/script-canvas/")
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &EditorScriptCanvasComponent::m_sourceHandle, "Script Canvas Source File", "Script Canvas source file associated with this component")
-                        ->Attribute("BrowseIcon", ":/stylesheet/img/UI20/browse-edit-select-files.svg")
-                        ->Attribute("EditButton", "")
-                        ->Attribute("EditDescription", "Open in Script Canvas Editor")
-                        ->Attribute("EditCallback", &EditorScriptCanvasComponent::OpenEditor)
-                        ->Attribute(AZ::Edit::Attributes::AssetPickerTitle, "Script Canvas")
-                        ->Attribute(AZ::Edit::Attributes::SourceAssetFilterPattern, "*.scriptcanvas")
-                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorScriptCanvasComponent::OnFileSelectionChanged)
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &EditorScriptCanvasComponent::m_variableOverrides, "Properties", "Script Canvas Graph Properties")
-                        ->Attribute(AZ::Edit::Attributes::Visibility, AZ::Edit::PropertyVisibility::ShowChildrenOnly)
-                    ;
-            }
-        }
-
-        if (AZ::JsonRegistrationContext* jsonContext = azrtti_cast<AZ::JsonRegistrationContext*>(context))
-        {
-            jsonContext->Serializer<AZ::EditorScriptCanvasComponentSerializer>()->HandlesType<EditorScriptCanvasComponent>();
-        }
-    }
-
     EditorScriptCanvasComponent::EditorScriptCanvasComponent()
```

Comment:
The EditorScriptCanvas component is now wildly simple, and is better reviewed be examining the file directly. It does nothing but add a Configuration class to the reflected editor data (this allows the user to select the script and configure properties in the editor), and implement BuildGameEntity by asking the BuilderDataSystem for the latest compiled data to hand over to the RuntimeComponent (it used to compile the data itself).